src/software/ai/hl/stp/play/ball_placement/ball_placement_play_fsm.h

```diff
@@ -0,0 +1,145 @@
+     * position with a kick
+     *
+     * @param event the BallPlacementPlayFSM Update event
+     *
+     * @return whether the kick has been performed
+     */
+    bool kickDone(const Update& event);
+
+    /**
+     * Guard on whether the placing robot has finished placing the ball into the desired
+     * position
+     *
+     * @param event the BallPlacementPlayFSM Update event
+     *
+     * @return whether the kick has been performed
+     */
+    bool ballPlaced(const Update& event);
+
+    /**
+     * Helper function for calculating the angle to kick the ball off of a wall
+     *
+     * @param ball_pos
+     * @param field_lines
+     *
+     * @return the kick angle
+     */
+    Angle calculateWallKickoffAngle(const Point& ball_pos, const Rectangle& field_lines);
+
+    /**
+     * Helper function for setting up the MoveTactics of the robots away from the ball
+     * placing robot
+     *
+     * @param event the BallPlacementPlayFSM Update event
+     */
+    void setupMoveTactics(const Update& event);
+
+    auto operator()()
+    {
+        using namespace boost::sml;
+
+        DEFINE_SML_STATE(StartState)
+        DEFINE_SML_STATE(KickOffWallState)
+        DEFINE_SML_STATE(PlaceBallState)
+        DEFINE_SML_STATE(RetreatState)
+        DEFINE_SML_EVENT(Update)
+
+        DEFINE_SML_ACTION(placeBall)
+        DEFINE_SML_ACTION(kickOffWall)
+        DEFINE_SML_ACTION(retreat)
+
+        DEFINE_SML_GUARD(shouldKickOffWall)
+        DEFINE_SML_GUARD(kickDone)
+        DEFINE_SML_GUARD(ballPlaced)
+
+        return make_transition_table(
+            // src_state + event [guard] / action = dest_state
+            *StartState_S + Update_E[shouldKickOffWall_G]       = KickOffWallState_S,
+            *StartState_S + Update_E[!shouldKickOffWall_G]      = PlaceBallState_S,
+            KickOffWallState_S + Update_E[!shouldKickOffWall_G] = PlaceBallState_S,
+            KickOffWallState_S + Update_E[!kickDone_G] / kickOffWall_A =
+                KickOffWallState_S,
+            KickOffWallState_S + Update_E[kickDone_G]                = StartState_S,
+            PlaceBallState_S + Update_E[shouldKickOffWall_G]         = StartState_S,
+            PlaceBallState_S + Update_E[!ballPlaced_G] / placeBall_A = PlaceBallState_S,
+            PlaceBallState_S + Update_E[ballPlaced_G] / retreat_A    = RetreatState_S,
+            RetreatState_S + Update_E[!ballPlaced_G]                 = StartState_S,
+            RetreatState_S + Update_E[ballPlaced_G] / retreat_A      = X,
+            X + Update_E[!ballPlaced_G]                              = StartState_S);
+    }
+
+   private:
+    TbotsProto::AiConfig ai_config;
+    std::shared_ptr<WallKickoffTactic> pivot_kick_tactic;
+    std::shared_ptr<PlaceBallTactic> place_ball_tactic;
+    std::shared_ptr<MoveTactic> retreat_tactic;
+    std::vector<std::shared_ptr<PlaceBallMoveTactic>> move_tactics;
+    static double const shot_velocity_threshold;
```

Comment:
this syntax is a little odd so I'm not too sure if you can reassign a variable declared const later successfully in the cpp file. I suggest:

```suggestion
    static double const SHOT_VELOCITY_THRESHOLD_M_PER_S = 1.0;
    static double const WALL_KICKOFF_THRESHOLD_M_PER_S = 3.0;
```